File: protocol/api/src/main/java/ink/reactor/protocol/api/packet/PacketHandlerStorage.java
package ink.reactor.protocol.api.packet;

import ink.reactor.protocol.api.connection.PlayerConnection;
import ink.reactor.protocol.api.buffer.reader.ReaderBuffer;

public interface PacketHandlerStorage {
    void callHandlers(final PlayerConnection connection, final ReaderBuffer buffer, final int packetID);

    void add(final PacketHandler handler);
    void remove(final PacketHandler handler);

    void add(final PacketHandler... handlers);
    void remove(final PacketHandler... handlers);

    void clear(final int id);
}

File: networking/api/src/main/kotlin/ink/reactor/network/api/packet/handler/PacketHandlerStorage.kt
package ink.reactor.network.api.packet.handler

import ink.reactor.network.api.packet.Packet
import ink.reactor.network.api.player.PlayerConnection

interface PacketHandlerStorage {
    fun callHandlers(connection: PlayerConnection, packet: Packet)

    fun add(handler: PacketHandler)
    fun remove(handler: PacketHandler)

    fun add(vararg handlers: PacketHandler)
    fun remove(vararg handlers: PacketHandler)

    fun clear(id: Int)
}
